spec/classes/server/conf_spec.rb

```diff
@@ -249,6 +253,17 @@
           it { is_expected.to contain_class('haveged') }
         end
 
+        context 'when connected to an IPA domain' do
+          let(:pre_condition){ 'include "::ssh"' }
+          let(:facts) {
+            os_facts.merge(
+              ipa: { 'status' => 'joined' }
+            )
```

Comment:
Once the ipa fact is fixed (PR to be merged), 'status' is no longer a valid key.  So, you may want to either have an empty hash for ipa (you don't care what is in the hash) or have the  'domain', and 'server' keys as you did in host_config_entry_spec.rb